Content.Server/DeltaV/StationEvents/Events/PirateRadioSpawnRule.cs

```diff
@@ -1,32 +1,91 @@
 using Robust.Server.GameObjects;
 using Robust.Server.Maps;
 using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Utility;
 using Content.Server.GameTicking;
-using Content.Server.GameTicking.Rules;
 using Content.Server.GameTicking.Rules.Components;
 using Content.Server.StationEvents.Components;
-using Content.Server.RoundEnd;
+using Content.Server.Station.Components;
+using Content.Shared.Salvage;
+using Content.Shared.Random.Helpers;
+using System.Linq;
 
 namespace Content.Server.StationEvents.Events;
 
 public sealed class PirateRadioSpawnRule : StationEventSystem<PirateRadioSpawnRuleComponent>
 {
+    [Dependency] private readonly IEntityManager _entities = default!;
     [Dependency] private readonly IMapManager _mapManager = default!;
     [Dependency] private readonly MapLoaderSystem _map = default!;
-    [Dependency] private readonly GameTicker _gameTicker = default!;
-    [Dependency] private readonly TraitorRuleSystem _TraitorRuleSystem = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
 
     protected override void Started(EntityUid uid, PirateRadioSpawnRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
     {
+        //This handles spawning the Listening Outpost
         base.Started(uid, component, gameRule, args);
-		
-        var shuttleMap = _mapManager.CreateMap();
-        var options = new MapLoadOptions
+
+        var xformQuery = GetEntityQuery<TransformComponent>();
+        //Find where the station is and get a bounding box
+        var aabbs = EntityQuery<StationDataComponent>().SelectMany(x =>
+                x.Grids.Select(x =>
+                    xformQuery.GetComponent(x).WorldMatrix.TransformBox(_mapManager.GetGridComp(x).LocalAABB)))
+            .ToArray();
+        if (aabbs.Length < 1) return;
+        var aabb = aabbs[0];
+
+        for (var i = 1; i < aabbs.Length; i++)
+        {
+            aabb.Union(aabbs[i]);
+        }
+        //Generates a potential spawning area, shaped like a square with rounded corners.
+        //DistanceModifier allows for fine tuning of how large this area is.
+        //The initial target distance(for DistanceModifier = 20f) is between 1km and 1.5km of the station.
+        //But with the way NextVector2 works, it can be less than that
+        //Do not for any reason set DistanceModifier greater than 25f or less than 1f. 
```

Comment:
Then make this a property on the component (the get, set stuff) and clamp it. Unless that's out of fashion now 
![image](https://github.com/DeltaV-Station/Delta-v/assets/60792108/13ac891e-0e0b-4394-b338-ff7764aa1720)